### DISCUSSION TITLE
More Flexible tag based filtering for Annotations on Dashboards

### DISCUSSION THREAD

**Discussion by martinrw** [grafana/grafana#72999] [Dashboards, Panels & Visualizations]
We have a use case where we want a combination of Optional + required tags in the tag filter for annotations on one of our dashboards.

In our example the annotation has tags like this (using key value pairs):
type: release, service:my_service, team:my_team
type: rollback, service:my_service, team: my_team
These annotations aren't linked to a particular dashboard.

now in our dashboard I want to apply a filter that looks something like:
Always Match: "type: release"
Match any: "service:$selected_service"

where $selected_service is defined by a variable which could be a multi or all select.

so basically this means we can see all releases but then choose to filter it down any number of services.

The problem I currently am running into is:
If I choose the option "match any" then the "type:release" match will cause us to match all services and the service based filter doesn't do anything
or, if I keep "match any" unselected and $selected_service is multiple services or all (.*) it doesn't match anything

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82547. Feel free to continue the discussion around this there. Thank you!
